ext/date/php_date.c

```diff
@@ -2832,6 +2837,34 @@ PHP_METHOD(DateTime, __wakeup)
 }
 /* }}} */
 
+/* {{{ proto DateTime::createFromImmutable(DateTimeImmutable object)
+   Creates new DateTime object from an existing DateTimeImmutable object.
+*/
+PHP_METHOD(DateTime, createFromImmutable)
+{
+	zval *datetimeimmutable_object = NULL;
+	php_date_obj *new_obj = NULL;
+	php_date_obj *old_obj = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O!", &datetimeimmutable_object, date_ce_immutable) == FAILURE) {
+		return;
+	}
+
+	php_date_instantiate(date_ce_date, return_value);
+	old_obj = Z_PHPDATE_P(datetimeimmutable_object);
+	new_obj = Z_PHPDATE_P(return_value);
+
+	new_obj->time = timelib_time_ctor();
+	*new_obj->time = *old_obj->time;
```

Comment:
Um, isn't the previous line calling the constructor redundant, since you're just assigning over it in the following line? Isn't that a memory leak?